Warships/MainWindow.xaml.cs

```diff
@@ -56,31 +72,56 @@ private void CreateBoard()
             }
         }
         
-        private void OnMouseEnter(object sender, MouseEventArgs e)
+        private void OnMouseEnter(object sender, MouseEventArgs e, int id)
         {
-            if (e.OriginalSource is Rectangle activeRectangle)
+            if (e.OriginalSource is not Rectangle) return;
+            
+            var height = id / _board.Width;
+            var width = id % _board.Width;
```

Comment:
I would call this `x` and `y` instead of height and width, as this are specific coordinates

Reply:
Same in `OnMouseLeave` and `OnMouseLeftButtonDown`